### ISSUE TITLE
Cell dimensions?

### ISSUE THREAD

**Issue by Fil** [observablehq/framework#933]
I want SSR (#931), but it seems like a distant goal: it's not easy to do from node.js without strong constraints that make the code work both in Node and in the browser, not always possible (e.g. live data), and not always useful to SSR (for example, a complex image generated by a simple algorithm might be much heavier to download than the generating code).

However, and somewhat independently, we should be able to register the _dimensions_ of unrendered cells (width, height, and maybe a few other computed styles) as a cheap way to eliminate "Cumulative Layout Shift". CLS —stuff moving around as the page loads— which is the one metric that gives Framework bad grades in terms of performance and reader experience.

I could see this as a manual process (adding ```js height=500px), though I suspect some simple automation might be possible without necessarily having to do a full SSR.

**Comment by mbostock** [comment]
Yeah. We should also support expressing the height as an aspect ratio since in many cases content is responsive. Of course in the case of text wrapping it’s impossible to server-side render the dimensions accurately, so it may be a fool’s errand to attempt this and we do need actual SSR…
